lib/core/src/Cardano/Wallet.hs

```diff
@@ -1439,14 +1443,21 @@ selectAssets ctx (utxo, cp, pending) tx outs transform = do
     liftIO $ traceWith tr $ MsgSelectionStart utxo outs
     selectionCriteria <- withExceptT ErrSelectAssetsCriteriaError $ except $
         initSelectionCriteria tl pp tx utxo outs
-    sel <- performSelection
+    mSel <- performSelection
         (view #txOutputMinimumAdaQuantity $ constraints tl pp)
         (calcMinimumCost tl pp tx)
         (tokenBundleSizeAssessor tl)
         (selectionCriteria)
-    liftIO $ traceWith tr $ MsgSelectionDone sel
+    case mSel of
```

Comment:
It is possible to just trace `Either e SelectionResult` and perform the reporting at the time of log formatting, and/or do further tracer transforms. That way, your functions aren't cluttered up with logging code.